src/client/components/element-info/entity-info.js

```diff
@@ -148,6 +148,23 @@ class EntityInfo extends DataComponent {
       delete match.typeOfGene; // n.b. the model overrides via this field
     }
 
+    // If label is different from match 'name', add latter to shortSynonyms
+    const label = s.name;
+    let matchShortSyns = _.get( match, 'shortSynonyms', [] );
+    let inputNotMatchName = label !== match.name;
+    if ( inputNotMatchName ) {
+      let newShortSyns = _.uniq( _.concat( match.name, matchShortSyns ) );
+      _.set( match, 'shortSynonyms', newShortSyns );
+    }
+
+    // Remove the match 'name' from the current association, if exists
+    const assoc = el.association();
```

Comment:
Since the association will be replaced on line `el.associate( match );`, should this instead modify `match`?

Reply:
My thinking was: `assoc` will be unassociated when the `match` is associated with `el`, so you need to reset it to its original state (drop the name from short synonyms).

Reply:
That makes sense.  I don't think an association is ever reused, though, so it probably doesn't matter.  As long as this wasn't intended for the incoming association, then I think it's good to merge.